Sources_v3/Query/Pagination.swift

```diff
@@ -4,67 +4,60 @@
 
 import Foundation
 
-public typealias Pagination = Set<PaginationOption>
+public extension Int {
+    /// A default channels page size.
+    static let channelsPageSize = 20
+    /// A default channels page size for the next page.
+    static let channelsNextPageSize = 30
+    /// A default messages page size.
+    static let messagesPageSize = 25
+    /// A default messages page size for the next page.
+    static let messagesNextPageSize = 50
+    /// A default users page size.
+    static let usersPageSize = 30
+    /// A default channel members page size.
+    static let channelMembersPageSize = 30
+}
 
-public extension Pagination {
-    var limit: Int? {
-        first(where: { $0.limit != nil })?.limit
+public struct Pagination: Encodable, Equatable {
```

Comment:
I think we have two "versions" of pagination and we should even distinguish them type-wise (?):
- one should have `pageSize` value and optional `offset`
- the other has optional page size and accepts a set of `greaterThan` `greaterThanOrEqual` `lessThan` `lessThanOrEqual`

Some queries support one and some support the other.